diff --git a/servo_pkg/include/servo_pkg/servo_mgr.hpp b/servo_pkg/include/servo_pkg/servo_mgr.hpp
index a14fb5d..51cc5b8 100644
--- a/servo_pkg/include/servo_pkg/servo_mgr.hpp
+++ b/servo_pkg/include/servo_pkg/servo_mgr.hpp
@@ -44,7 +44,7 @@ namespace PWM {
     #define POLARITY_MOTOR_VAL -1
     #define SERVO_PERIOD 20000000
     /// Servo GPIO channel, set by Pega
-    #define GPIO_CHANNEL 436
+    #define GPIO_CHANNEL 1
 
     class ServoMgr
         {
