diff --git a/inference_pkg/src/inference_node.cpp b/inference_pkg/src/inference_node.cpp
index 12628fd..672d3f5 100644
--- a/inference_pkg/src/inference_node.cpp
+++ b/inference_pkg/src/inference_node.cpp
@@ -53,7 +53,7 @@ namespace InferTask {
             // Device name; OpenVINO supports CPU, GPU and MYRIAD
             deviceName_ = this->get_parameter("device").as_string();
 
-            loadModelServiceCbGrp_ = this->create_callback_group(rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
+            loadModelServiceCbGrp_ = this->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
             loadModelService_ = this->create_service<deepracer_interfaces_pkg::srv::LoadModelSrv>("load_model",
                                                                                                   std::bind(&InferTask::InferenceNodeMgr::LoadModelHdl,
                                                                                                   this,
@@ -63,7 +63,7 @@ namespace InferTask {
                                                                                                   ::rmw_qos_profile_default,
                                                                                                   loadModelServiceCbGrp_);
 
-            setInferenceStateServiceCbGrp_ = this->create_callback_group(rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
+            setInferenceStateServiceCbGrp_ = this->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
             setInferenceStateService_ = this->create_service<deepracer_interfaces_pkg::srv::InferenceStateSrv>("inference_state",
                                                                                                                std::bind(&InferTask::InferenceNodeMgr::InferStateHdl,
                                                                                                                this,
@@ -143,11 +143,11 @@ namespace InferTask {
 
     private:
         /// ROS callback group for load model service.
-        rclcpp::callback_group::CallbackGroup::SharedPtr loadModelServiceCbGrp_;
+        rclcpp::CallbackGroup::SharedPtr loadModelServiceCbGrp_;
         /// ROS service to load inference model.
         rclcpp::Service<deepracer_interfaces_pkg::srv::LoadModelSrv>::SharedPtr loadModelService_;
         /// ROS callback group for set inference state service.
-        rclcpp::callback_group::CallbackGroup::SharedPtr setInferenceStateServiceCbGrp_;
+        rclcpp::CallbackGroup::SharedPtr setInferenceStateServiceCbGrp_;
         /// ROS service to set the inference state to start/stop running inference.
         rclcpp::Service<deepracer_interfaces_pkg::srv::InferenceStateSrv>::SharedPtr setInferenceStateService_;
 
