diff --git a/servo_pkg/include/servo_pkg/servo_mgr.hpp b/servo_pkg/include/servo_pkg/servo_mgr.hpp
index a14fb5d..51cc5b8 100644
--- a/servo_pkg/include/servo_pkg/servo_mgr.hpp
+++ b/servo_pkg/include/servo_pkg/servo_mgr.hpp
@@ -44,7 +44,7 @@ namespace PWM {
     #define POLARITY_MOTOR_VAL -1
     #define SERVO_PERIOD 20000000
     /// Servo GPIO channel, set by Pega
-    #define GPIO_CHANNEL 436
+    #define GPIO_CHANNEL 1
 
     class ServoMgr
         {
diff --git a/servo_pkg/src/servo_node.cpp b/servo_pkg/src/servo_node.cpp
index 631af08..1a88136 100644
--- a/servo_pkg/src/servo_node.cpp
+++ b/servo_pkg/src/servo_node.cpp
@@ -62,7 +62,7 @@ int main(int argc, char **argv) {
                                                                                                  std::placeholders::_1),
                                                                                         rclcpp::SubscriptionOptions(),
                                                                                         rawPWMMsgStrategy);
-    auto servoCalServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto servoCalServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::SetCalibrationSrv>::SharedPtr servoCalService =
         node->create_service<deepracer_interfaces_pkg::srv::SetCalibrationSrv>(SET_CAL_SERVICE_NAME,
                                                                                std::bind(&PWM::ServoMgr::setCalHdl,
@@ -73,7 +73,7 @@ int main(int argc, char **argv) {
                                                                                ::rmw_qos_profile_default,
                                                                                servoCalServiceCbGrp_);
 
-    auto setGPIOServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);                                                                                          
+    auto setGPIOServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
     rclcpp::Service<deepracer_interfaces_pkg::srv::ServoGPIOSrv>::SharedPtr setGPIOService =
         node->create_service<deepracer_interfaces_pkg::srv::ServoGPIOSrv>(SERVO_GPIO_SERVICE_NAME,
                                                                           std::bind(&PWM::ServoMgr::setGPIOHdl,
@@ -84,7 +84,7 @@ int main(int argc, char **argv) {
                                                                           ::rmw_qos_profile_default,
                                                                           setGPIOServiceCbGrp_);
 
-    auto getCalServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto getCalServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::GetCalibrationSrv>::SharedPtr getCalService =
         node->create_service<deepracer_interfaces_pkg::srv::GetCalibrationSrv>(GET_CAL_SERVICE_NAME,
                                                                        std::bind(&PWM::ServoMgr::getCalHdl,
@@ -95,7 +95,7 @@ int main(int argc, char **argv) {
                                                                        ::rmw_qos_profile_default,
                                                                        getCalServiceCbGrp_);
 
-    auto setLedCtrlCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto setLedCtrlCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::SetLedCtrlSrv>::SharedPtr setLedCtrlService =
         node->create_service<deepracer_interfaces_pkg::srv::SetLedCtrlSrv>(SET_LED_STATE_SERVICE_NAME,
                                                                            std::bind(&PWM::LedMgr::setLedCtrlHdl,
@@ -106,7 +106,7 @@ int main(int argc, char **argv) {
                                                                            ::rmw_qos_profile_default,
                                                                            setLedCtrlCbGrp_);
 
-    auto getLedCtrlCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto getLedCtrlCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::GetLedCtrlSrv>::SharedPtr getLedCtrlService =
         node->create_service<deepracer_interfaces_pkg::srv::GetLedCtrlSrv>(GET_LED_STATE_SERVICE_NAME,
                                                                            std::bind(&PWM::LedMgr::getLedCtrlHdl,
