diff --git a/status_led_pkg/status_led_pkg/constants.py b/status_led_pkg/status_led_pkg/constants.py
index 9c3676a..be97c1e 100644
--- a/status_led_pkg/status_led_pkg/constants.py
+++ b/status_led_pkg/status_led_pkg/constants.py
@@ -27,9 +27,9 @@ DEFAULT_LED_INDEX = 0
 # Status light LED GPIO port matrix.
 # Cols: led indices; rows: r, g, b channel ports.
 LED_PORTS = (
-    (448, 447, 437),
-    (446, 445, 443),
-    (450, 457, 458)
+    (488+7, 488+8, 488+9),
+    (488+10, 488+11, 488+12),
+    (488+13, 488+14, 488+15),
 )
 
 
