diff --git a/model_optimizer_pkg/model_optimizer_pkg/model_optimizer_node.py b/model_optimizer_pkg/model_optimizer_pkg/model_optimizer_node.py
index ce5b7bd..90277d9 100644
--- a/model_optimizer_pkg/model_optimizer_pkg/model_optimizer_node.py
+++ b/model_optimizer_pkg/model_optimizer_pkg/model_optimizer_node.py
@@ -355,17 +355,23 @@ class ModelOptimizerNode(Node):
 
 
 def main(args=None):
-   
-    rclpy.init(args=args)
-    model_optimizer_node = ModelOptimizerNode()
-    executor = MultiThreadedExecutor()
-    rclpy.spin(model_optimizer_node, executor)
-    # Destroy the node explicitly
-    # (optional - otherwise it will be done automatically
-    # when the garbage collector destroys the node object)
-    model_optimizer_node.destroy_node()
-    rclpy.shutdown()
 
+    try:
+        rclpy.init(args=args)
+        model_optimizer_node = ModelOptimizerNode()
+        executor = MultiThreadedExecutor()
+        rclpy.spin(model_optimizer_node, executor)
+        # Destroy the node explicitly
+        # (optional - otherwise it will be done automatically
+        # when the garbage collector destroys the node object)
+        model_optimizer_node.destroy_node()
+
+    except KeyboardInterrupt:
+        pass
+
+    finally:
+        if rclpy.ok():
+            rclpy.shutdown()
 
 if __name__ == "__main__":
     main()
